src/main/java/com/dataguise/saas/service/S3LfaService.java

```diff
@@ -0,0 +1,207 @@
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return bucketsDirInfo;
+    }
+
+    /**
+     * Method used for creationg the LFA-S3 cluster
+     * @param dgAgentClusterDTO
+     * @return
+     */
+    public Integer saveCluster(DgAgentClusterDTO dgAgentClusterDTO) throws BadGatewayException, BadRequestException{
+        Integer clusterId=null;
+        try {
+            DgClusterInfo dgClusterInfo = createClusterInfo(dgAgentClusterDTO,0);
+            clusterId = restDriver.saveClusterInfo(dgClusterInfo);
+
+        } catch (DgDrvException e) {
+            logger.error("create cluster failed " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return clusterId;
+    }
+    /**
+     * Method is used for creating cluster info bean i.e. used in the create cluster call
+     * @param dgAgentClusterDTO
+     * @return
+     */
+
+    private DgClusterInfo createClusterInfo(DgAgentClusterDTO dgAgentClusterDTO, Integer id) {
+        DgAgentClusterInfo dgAgentClusterInfo = new DgAgentClusterInfo();
+        DgCloudAgentInfo dgCloudAgentInfo = new DgCloudAgentInfo();
+        dgAgentClusterInfo.setIsPrimary(true);
+        dgAgentClusterInfo.setIsSecondary(false);
+        DgClusterInfo dgClusterInfo = new DgClusterInfo();
+        dgClusterInfo.setId(id);
+        dgClusterInfo.setClusterType(Modules.S3LFA.getModuleType());
+        dgClusterInfo.setLocation("Cloud");
+        dgClusterInfo.setClusterName(dgAgentClusterDTO.getClusterName());
+
+
+        List<DgAgentClusterInfo> dgAgentClusterInfoList = new ArrayList<>();
+        List<DgCloudAgentInfo> dgCloudAgentInfoList = new ArrayList<>();
+
+        // set details for LFA Agent
+        dgAgentClusterInfo.setAgentId(dgAgentClusterDTO.getClusterAgentId());
+        dgAgentClusterInfoList.add(dgAgentClusterInfo);
+
+        // set details for S3 Agent
+        dgCloudAgentInfo.setAgentId(dgAgentClusterDTO.getCloudAgentId());
+        dgCloudAgentInfoList.add(dgCloudAgentInfo);
+
+        dgClusterInfo.setAgentInfo(dgAgentClusterInfoList);
+        dgClusterInfo.setCloudAgentInfo(dgCloudAgentInfoList);
+        return dgClusterInfo;
+    }
+
+    /**
+     * Method is used for editing the cluster Info using ID
+     * @param dgAgentClusterDTO
+     * @return
+     * @throws BadGatewayException
+     * @throws BadRequestException
+     */
+    public Integer editClusterInfo(DgAgentClusterDTO dgAgentClusterDTO, Integer id) throws BadGatewayException, BadRequestException{
+        Integer clusterId=null;
+        try {
+            DgClusterInfo dgClusterInfo = createClusterInfo(dgAgentClusterDTO, id);
+            clusterId = restDriver.editClusterInfo(dgClusterInfo,id);
+
+        } catch (DgDrvException e) {
+            logger.error("create cluster failed " + e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/S3LfaService.java L119
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1580572258>*